src/Utils/BlockTemplateUtils.php

```diff
@@ -6,6 +6,42 @@
  * IMPORTANT: These methods have been duplicated from Gutenberg/lib/full-site-editing/block-templates.php as those functions are not for public usage.
  */
 class BlockTemplateUtils {
+	/**
+	 * Old directory name of the block templates directory.
+	 *
+	 * The convention has changed with Gutenberg 12.1.0, but we need to keep this
+	 * for backwards compatibility.
+	 *
+	 * @deprecated
+	 * @var string
+	 */
+	const DEPRECATED_TEMPLATES_DIR_NAME = 'block-templates';
+
+	/**
+	 * Old directory name of the block template parts directory.
+	 *
+	 * The convention has changed with Gutenberg 12.1.0, but we need to keep this
+	 * for backwards compatibility.
+	 *
+	 * @deprecated
+	 * @var string
+	 */
+	const DEPRECATED_TEMPLATE_PARTS_DIR_NAME = 'block-templates-parts';
+
+	/**
+	 * Directory name of the block templates directory.
+	 *
+	 * @var string
+	 */
+	const TEMPLATES_DIR_NAME = 'templates';
+
+	/**
+	 * Directory name of the block template parts directory.
+	 *
+	 * @var string
+	 */
+	const TEMPLATE_PARTS_DIR_NAME = 'parts';
```

Comment:
Can we not put these values in an [associative array](https://www.php.net/manual/en/language.types.array.php)?

Reply:
We 100% can. I was just following the pre-existing convention. Also, in the spirit of trying to avoid as many side-effects as possible, not changing that convention makes the PR slightly safer.

But I can convert those values in an associative array, if you so prefer.

Reply:
I think it might make sense to put these in an associative array, I don't think it'll be too risky given the convention hasn't been set in this file.